### ISSUE TITLE
run-all should halt when a fetch cell has an error

### ISSUE THREAD

**Issue by bcolloran** [iodide-project/iodide#1112]

**Comment by hamilton** [comment]
This will require a bit of refactoring how we handle evaluation. Currently, it looks something like this:

```
editor sends 10 code chunks
eval frame turns each chunk into an action in a promise chain
chunk 1: eval (success)
chunk 2: eval (fail, report in console, no way to reset eval queue)
chunk 3: eval (success)
chunk 4: fail (fails because chunk 2 fails)
```

It needs to be more like this:

```
editor sends code chunk 1
eval on chunk 1 -> report success to editor
editor sends code chunk 2
eval on chunk 2 -> FAIL, report to editor
editor blocks other code chunks
```

**Comment by iFlameing** [comment]
@hamilton  can you explain how I can proceed regarding this issue.

**Comment by bcolloran** [comment]
@hamilton did your eval queue work fix this?

**Comment by bcolloran** [comment]
it looks like the fetch cell erroring does not halt the eval queue, but any other errors do; updating issue to focus on that

**Comment by hamilton** [comment]
@bcolloran this is essentially a duplicate of #1372, and will be fixed by #1410. Will close in favor of the other issue
